test/factories/users.rb

```diff
@@ -20,5 +20,12 @@
         Array.new(1).map { |_n| create(:department) }
       end
     end
+
+    # after(:create) do |user, profile|
+    #   user.profile
+    #   user.reload
+    #   # u.profile = create(:profile, user: u)
+    #   # u.profile.contact = create(:contact, contactable: u)
+    # end
```

Comment:
You can try `after(:build)`, IIRC that would get called before the record is saved, as opposed to `after(:create)` which gets called after saving (which I assume causes the validation errors).

Or you can use the `after_initialize` approach from #100 then it should happen automatically.